test/functional/search/search-content-check-positive.bats

```diff
@@ -57,12 +57,13 @@ global_teardown() {
 
 	assert_status_is 0
 	expected_output=$(cat <<-EOM
+		Downloading all Clear Linux manifests
 		Searching for 'test-bundle1'
 		Bundle test-bundle1 \\(0 MB to install\\)
 		./usr/share/clear/bundles/test-bundle1
 	EOM
 	)
-	assert_regex_is_output "$expected_output"
+	assert_regex_in_output "$expected_output"
```

Comment:
I think that a validation using in_output here is checking better the expected from this search-file than is_output.
By using in_output we are being specific on what we are testing, that in this case is that the bundles were located correctly and in correct order.
We use is_output in most of our tests and this makes tests harder to maintain, because they need to be update if there is any change in output in any other part of the code that is executed by this function. And I'm not convinced that a is_output check in all tests are really testing more than using in_output.
In this specific case the problem is knowing when the cache is or isn't populated to add or not a string "Downloading all Clear Linux manifests" before running the test.
It's possible to do it, but I think it's going to be hard to maintain.

Do you think we are losing something on using in_output here?

Reply:
oh I see, no I think in this case is ok to do it like this.

But, besides special cases like this, in general I do think is better to explicitly use is_output over in_output, I know it makes tests a little bit harder to maintain (not really "harder" to maintain, just more boring to maintain), but that way we make sure the output looks the way we like, if you only tests part of the output in each test you could end up with an output that makes no sense or looks ugly when looking it overall. I'm always very picky on how output looks because the quality of a product is only as good as it is perceived to be by the user, and the output makes a big part of that. Before testlib was in place we were doing it that way, testing only specific parts of the output, and we had many repetitive messages, like multiple warnings for the same thing, etc.